terraform-v2/v1-to-v2.md

```diff
@@ -0,0 +1,89 @@
+# Upgrading From Terraform Orb V1 To V2
+
+## Why V2?
+
+Version management tools such as `tfswitch` and `tfenv`, alongside the
+`required_version` terraform config setting, now seem to be relatively
+common-place, and remove the need for new Docker image/executors to be curated
+for each minor version of terraform.
+
+We’d like to keep the orb source code as simple as possible, whilst implementing
+new Terraform flags/features. Having different code paths based on the Terraform
+version has been steadily increasing the required complexity.
+
+We are taking advantage of some removals in terraform `0.15` which
+break the terraform v1 orb. We would like to fix those removals in v2 only,
+resulting in a situation where the terraform v1 orb supports terraform `<=0.14`
+and terraform v2 supports terraform `>=0.14`.
+
+## How Has V2 Been Split From V1?
+
+In an attempt to continue supporting orb v1 in the short term, we’ve created an
+entirely new orb called “terraform-v2”. We can maintain both major versions
+from the same codebase.
+
+Upgrading is therefore entirely in the control of the user, no-one will be 
+automatically upgraded.
+
+## Pre-requisites To Upgrading
+
+Before upgrading, you must ensure you meet the following criteria:
+
+- Running terraform >=0.14
+- Managing terraform version using `tfswitch`, `tfenv` or `required_version`
+- If running Helm commands in the container prior to any terraform commands,
+ensure you're using Helm3, not Helm2
+- Ensure you're not using the `use_chdir` orb parameter (introduced as an
+optional parameter in orb v1 in version `1.8.3`)
+
+## How To Upgrade From V1 to V2
+
+You'll need to update the orb stanza in your .circleci/config.yml file:
+
+```yaml
+orbs:
+  terraform: ovotech/terraform-v2@1
+```
+
+This will use the latest 1.x.x version of the v2 orb, you can pin to a more
```

Comment:
Minor thing, but are we going with 1.x.x for the first version of v2 orb? Would 2.x.x be more intuitive?

Reply:
Yep I agree, fixed in dc0877d